packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -444,30 +444,46 @@ export default class SalesforceCRMService implements CRM {
           return [{ id: contact.Id, email: contact.Email }];
         }
 
-        await Promise.all(
-          contactsToCreate.map(async (attendee) => {
-            return await conn
-              .sobject(SalesforceRecordEnum.LEAD)
-              .create(
-                this.generateCreateRecordBody({
-                  attendee,
-                  recordType: SalesforceRecordEnum.LEAD,
-                  organizerId,
-                })
-              )
-              .then((result) => {
-                if (result.success) {
-                  createdContacts.push({ id: result.id, email: attendee.email });
-                }
-              });
-          })
-        );
+        for (const attendee of contactsToCreate) {
+          try {
+            const result = await conn.sobject(SalesforceRecordEnum.LEAD).create(
+              this.generateCreateRecordBody({
+                attendee,
+                recordType: SalesforceRecordEnum.LEAD,
+                organizerId,
+              })
+            );
+            if (result.success) {
+              createdContacts.push({ id: result.id, email: attendee.email });
+            }
+          } catch (error: any) {
+            console.error("Error creating lead:", error);
+            if (error.name === "DUPLICATES_DETECTED") {
+              const existingId = this.getExistingIdFromDuplicateError(error);
+              if (existingId) {
+                console.log("Using existing record:", existingId);
+                createdContacts.push({ id: existingId, email: attendee.email });
+              }
+            }
+          }
+        }
```

Comment:
I was getting this error 

`{"name":"DUPLICATES_DETECTED","duplicateResult":{"allowSave":true,"duplicateRule":"Standard_Lead_Duplicate_Rule","duplicateRuleEntityType":"Lead","errorMessage":"Use one of these records?","matchResults":[{"entityType":"Contact","errors":[],"matchEngine":"FuzzyMatchEngine","matchRecords":[{"additionalInformation":[],"fieldDiffs":[],"matchConfidence":100,"record":{"attributes":{"type":"Contact","url":"/services/data/v42.0/sobjects/Contact/0034w00004fYXCEAA4"},"Id":"0034w00004fYXCEAA4"}}],"rule":"Standard_Contact_Match_Rule_v1_1","size":1,"success":true}]},"errorCode":"DUPLICATES_DETECTED"}`

Reply:
So if there is duplicate record then we use the existing id from the error message